WcaOnRails/vendor/assets/javascripts/jquery.are-you-sure.js

```diff
@@ -68,13 +68,27 @@
       $field.data('ays-orig', getValue($field));
     };
 
-    var checkForm = function(evt) {
+    // Oftentimes, forms have some javascript that runs immediately when fields change.
+    // It makes sense to wait for those events to run (and possibly dirty or clean the form)
+    // before we check to see if the form is dirty. -JFLY
+    var checkForm = function(evt) {//JFLY
+      var that = this;
+      setTimeout(function() {//JFLY
+        checkFormNow.call(that, evt);//JFLY
+      }, 0);//JFLY
+    };//JFLY
+    var checkFormNow = function(evt) {
 
       var isFieldDirty = function($field) {
         var origValue = $field.data('ays-orig');
         if (undefined === origValue) {
           return false;
         }
+        if($field.is(":disabled")) {//JFLY
+          // Disabled fields cannot be dirty, as they won't
+          // submit anything to the server.
```

Comment:
I believe that under normal scenarios, disabled inputs cannot cause a change in the DB. Since browsers don't send over disabled inputs, you end up in a Rails controller and your `params` dict simply does not have a key => value entry in it. Then your controller calls `.update_attributes(safe_params)`, and the behavior of `update_attributes` is to ignore missing fields.

Does that sound right?